doc/releases/changelog-dev.md

```diff
@@ -7,6 +7,36 @@
 * Approximate Quantum Fourier Transform (AQFT) is now available from `qml.AQFT`.
   [(#4656)](https://github.com/PennyLaneAI/pennylane/pull/4656)
 
+* Iterative Quantum Phase Estimation is now available from `qml.iterative_qpe`.
+  [(#4804)](https://github.com/PennyLaneAI/pennylane/pull/4804)
+
+  The subroutine can be used similarly to mid-circuit measurements: 
+
+  ```python
+  
+  import pennylane as qml
+  
+  dev = qml.device("default.qubit", shots = 5)
+
+  @qml.qnode(dev)
+  def circuit():
+
+    # Initial state
+    qml.PauliX(wires = [0])
+
+    # Iterative QPE
+    measurements = qml.iterative_qpe(qml.RZ(2., wires = [0]), estimation_wire = [1], iters = 3)
+
+    return [qml.sample(op = meas) for meas in measurements]
+  ```
+  
+  ```pycon
+  >>> print(circuit())
+  [array([0, 0, 0, 0, 0]), array([1, 0, 0, 0, 0]), array([0, 1, 1, 1, 1])]
+  ```
+  
+  The i-th element in the list refers to the 5 samples generated by the i-th measurement of the algorithm.
```

Comment:
Now that I am seeing this, I wonder if this kind of return structure makes sense. Since we have 5 shots, I would think that it would be more intuitive if for each shot we had 3 measurement samples. This might apply to the whole of mid-circuit measurements. 

@mudit2812 @trbromley, what are your thoughts on this? Was this discussed in the PRD for MCMs?

Reply:
Yes, this is what I mean when I talk about general statistics. It is something that will be included soon. At that time it will be possible to do it:

`qml.sample(measurements)` and it will return what you comment 👌